protocol/x/clob/memclob/memclob.go

```diff
@@ -1307,6 +1307,24 @@ func (m *MemClobPriceTimePriority) validateNewOrder(
 		)
 	}
 
+	// Immediate-or-cancel orders may only be filled once. The remaining size becomes unfillable. This prevents
+	// the case where an IOC order is partially filled multiple times over the course of multiple blocks.
+	if order.TimeInForce == types.Order_TIME_IN_FORCE_IOC && remainingAmount < order.GetBaseQuantums() {
+		// Prevent IOC orders from replacing partially filled orders.
+		if restingOrderExists {
+			return errorsmod.Wrap(
+				types.ErrInvalidReplacement,
+				"Cannot replace partially filled order with IOC order",
```

Comment:
Suggestion: add more detail to the error (filled amount of current order probably)